### ISSUE TITLE
Unauthenticated access is not supported for this identity pool.

### ISSUE THREAD

**Issue by prabhukhanna** [awslabs/aws-lex-web-ui#5]
Hi, When I send a text message I receive a error from Cognito. The error is given below. 

**I was unable to process your message. NotAuthorizedException: Unauthenticated access is not supported for this identity pool.**

**Comment by atoa** [comment]
Hello,

It seems like your Cognito Identity Pool does not allow unauthenticated access. By default, this project uses an unauthenticated Identity Pool. You can create or configure the Identity Pool using the AWS Console or with a CloudFormation template provided in this repo (see instructions below). 

Using the the [Cognito console](https://console.aws.amazon.com/cognito) to allow unauthenticated access:

1. Go to **Manage Federated Identities**
2. Select the Identity Pool being used with the chatbot UI (or create one)
3. Click on **Edit Identity Pool**
4. Pick a role under the **Unauthenticated role** drop down (or create one). The IAM Policy associated with the selected role should allow the following actions: polly:SythensizeSpeech, lex:PostText and lex:PostContent
5. Expand **Unauthenticated Identities** and select **Enable access to unauthenticated identities**
6. Save changes

Alternatively, you can create a the Cognito Identity Pool and associated IAM role with this CloudFormation template: https://github.com/awslabs/aws-lex-web-ui/blob/master/templates/cognito.yaml. This template can automatically deploy the Cognito Identity pool with the required configuration and permissions. See the [Creating a Stack on the AWS CloudFormation Console](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-console-create-stack.html) for instruction on how to deploy a stack from a template.

Let me know if this solves this issue.

**Comment by prabhukhanna** [comment]
Thanks a lot. It worked. Whatever text(Including Buy flowers) I enter I am ending up with **Sorry, can you please repeat that?** Is there any configuration I am still missing ?

**Comment by atoa** [comment]
Excellent! Thanks for reaching out. Resolving this issue.

**Comment by nitinapexv1** [comment]
Dear Team
we are trying to use LEX project to configure SAML identity provider. 
we have created LEX authenticated role which 

{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "cognito-identity.amazonaws.com"
      },
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "StringEquals": {
          "cognito-identity.amazonaws.com:aud": "us-east-1:45fc1463-e2df-4415-94a0-0020c9b3495d"
        }
      }
    }
  ]
}

we have created SAML provider and attached to idendity pool
however we have not seen any interface goes to our ADFS instead it is giving error as below

aws-sdk-2.180.0.js:123726 POST https://cognito-identity.us-east-1.amazonaws.com/ 400 (Bad Request)
handleRequest @ aws-sdk-2.180.0.js:123726
executeSend @ aws-sdk-2.180.0.js:123142
SEND @ aws-sdk-2.180.0.js:123156
callListeners @ aws-sdk-2.180.0.js:127817
emit @ aws-sdk-2.180.0.js:127793
emit @ aws-sdk-2.180.0.js:126470
transition @ aws-sdk-2.180.0.js:125809
runTo @ aws-sdk-2.180.0.js:131045
(anonymous) @ aws-sdk-2.180.0.js:131057
(anonymous) @ aws-sdk-2.180.0.js:125825
(anonymous) @ aws-sdk-2.180.0.js:126472
callListeners @ aws-sdk-2.180.0.js:127831
emit @ aws-sdk-2.180.0.js:127793
emit @ aws-sdk-2.180.0.js:126470
transition @ aws-sdk-2.180.0.js:125809
runTo @ aws-sdk-2.180.0.js:131045
(anonymous) @ aws-sdk-2.180.0.js:131057
(anonymous) @ aws-sdk-2.180.0.js:125825
(anonymous) @ aws-sdk-2.180.0.js:126472
callListeners @ aws-sdk-2.180.0.js:127831
callNextListener @ aws-sdk-2.180.0.js:127811
(anonymous) @ aws-sdk-2.180.0.js:122986
(anonymous) @ aws-sdk-2.180.0.js:131801
hash @ aws-sdk-2.180.0.js:131534
sha256 @ aws-sdk-2.180.0.js:131485
computeSha256 @ aws-sdk-2.180.0.js:131799
COMPUTE_SHA256 @ aws-sdk-2.180.0.js:122980
callListeners @ aws-sdk-2.180.0.js:127817
emit @ aws-sdk-2.180.0.js:127793
emit @ aws-sdk-2.180.0.js:126470
transition @ aws-sdk-2.180.0.js:125809
runTo @ aws-sdk-2.180.0.js:131045
(anonymous) @ aws-sdk-2.180.0.js:131057
(anonymous) @ aws-sdk-2.180.0.js:125825
(anonymous) @ aws-sdk-2.180.0.js:126472
callListeners @ aws-sdk-2.180.0.js:127831
emit @ aws-sdk-2.180.0.js:127793
emit @ aws-sdk-2.180.0.js:126470
transition @ aws-sdk-2.180.0.js:125809
runTo @ aws-sdk-2.180.0.js:131045
(anonymous) @ aws-sdk-2.180.0.js:131057
(anonymous) @ aws-sdk-2.180.0.js:125825
(anonymous) @ aws-sdk-2.180.0.js:126472
callListeners @ aws-sdk-2.180.0.js:127831
emit @ aws-sdk-2.180.0.js:127793
emit @ aws-sdk-2.180.0.js:126470
transition @ aws-sdk-2.180.0.js:125809
runTo @ aws-sdk-2.180.0.js:131045
runTo @ aws-sdk-2.180.0.js:126190
send @ aws-sdk-2.180.0.js:126154
makeUnauthenticatedRequest @ aws-sdk-2.180.0.js:128167
getId @ aws-sdk-2.180.0.js:128613
getId @ aws-sdk-2.180.0.js:121131
refresh @ aws-sdk-2.180.0.js:121073
get @ aws-sdk-2.180.0.js:120814
(anonymous) @ aws-sdk-2.180.0.js:131873
promise @ aws-sdk-2.180.0.js:131872
getCredentials @ actions.js:633
wrappedActionHandler @ vuex.js:710
dispatch @ vuex.js:432
boundDispatch @ vuex.js:338
initCredentials @ actions.js:48
wrappedActionHandler @ vuex.js:710
dispatch @ vuex.js:432
boundDispatch @ vuex.js:338
(anonymous) @ LexWeb.vue:100
Promise.then (async)
created @ LexWeb.vue:99
callHook @ vue.js:2895
Vue._init @ vue.js:4547
VueComponent @ vue.js:4715
createComponentInstanceForVnode @ vue.js:4236
init @ vue.js:4053
createComponent @ vue.js:5499
createElm @ vue.js:5447
patch @ vue.js:5982
Vue._update @ vue.js:2637
updateComponent @ vue.js:2765
get @ vue.js:3113
Watcher @ vue.js:3102
mountComponent @ vue.js:2772
Vue$3.$mount @ vue.js:8416
Vue$3.$mount @ vue.js:10777
init @ vue.js:4059
createComponent @ vue.js:5499
createElm @ vue.js:5447
updateChildren @ vue.js:5736
patchVnode @ vue.js:5827
patch @ vue.js:5987
Vue._update @ vue.js:2647
updateComponent @ vue.js:2765
get @ vue.js:3113
run @ vue.js:3190
flushSchedulerQueue @ vue.js:2954
(anonymous) @ vue.js:1815
flushCallbacks @ vue.js:1736
Promise.then (async)
microTimerFunc @ vue.js:1784
nextTick @ vue.js:1828
queueWatcher @ vue.js:3041
update @ vue.js:3180
Vue.$forceUpdate @ vue.js:2668
forceRender @ vue.js:2292
(anonymous) @ vue.js:2302
(anonymous) @ vue.js:318
Promise.then (async)
resolveAsyncComponent @ vue.js:2326
createComponent @ vue.js:4145
_createElement @ vue.js:4357
createElement @ vue.js:4294
vm._c @ vue.js:4410
(anonymous) @ VM110:3
Vue._render @ vue.js:4465
updateComponent @ vue.js:2765
get @ vue.js:3113
Watcher @ vue.js:3102
mountComponent @ vue.js:2772
Vue$3.$mount @ vue.js:8416
Vue$3.$mount @ vue.js:10777
(anonymous) @ fullpage-component-loader.js:90
F @ _export.js:35
mountComponent @ fullpage-component-loader.js:72
(anonymous) @ fullpage-component-loader.js:43
Promise.then (async)
load @ fullpage-component-loader.js:42
(anonymous) @ index.js:95
Promise.then (async)
load @ index.js:94
load @ index.js:127
(anonymous) @ index.html:68
LexWeb.vue:128 could not initialize application while mounting: Error: Unauthenticated access is not supported for this identity pool.
    at Request.extractError (aws-sdk-2.180.0.js:124895)
    at Request.callListeners (aws-sdk-2.180.0.js:127821)
    at Request.emit (aws-sdk-2.180.0.js:127793)
    at Request.emit (aws-sdk-2.180.0.js:126470)
    at Request.transition (aws-sdk-2.180.0.js:125809)
    at AcceptorStateMachine.runTo (aws-sdk-2.180.0.js:131045)
    at aws-sdk-2.180.0.js:131057
    at Request.<anonymous> (aws-sdk-2.180.0.js:125825)
    at Request.<anonymous> (aws-sdk-2.180.0.js:126472)
    at Request.callListeners (aws-sdk-2.180.0.js:127831)
(anonymous) @ LexWeb.vue:128
Promise.catch (async)
created @ LexWeb.vue:98
callHook @ vue.js:2895
Vue._init @ vue.js:4547
VueComponent @ vue.js:4715
createComponentInstanceForVnode @ vue.js:4236
init @ vue.js:4053
createComponent @ vue.js:5499
createElm @ vue.js:5447
patch @ vue.js:5982
Vue._update @ vue.js:2637
updateComponent @ vue.js:2765
get @ vue.js:3113
Watcher @ vue.js:3102
mountComponent @ vue.js:2772
Vue$3.$mount @ vue.js:8416
Vue$3.$mount @ vue.js:10777
init @ vue.js:4059
createComponent @ vue.js:5499
createElm @ vue.js:5447
updateChildren @ vue.js:5736
patchVnode @ vue.js:5827
patch @ vue.js:5987
Vue._update @ vue.js:2647
updateComponent @ vue.js:2765
get @ vue.js:3113
run @ vue.js:3190
flushSchedulerQueue @ vue.js:2954
(anonymous) @ vue.js:1815
flushCallbacks @ vue.js:1736
Promise.then (async)
microTimerFunc @ vue.js:1784
nextTick @ vue.js:1828
queueWatcher @ vue.js:3041
update @ vue.js:3180
Vue.$forceUpdate @ vue.js:2668
forceRender @ vue.js:2292
(anonymous) @ vue.js:2302
(anonymous) @ vue.js:318
Promise.then (async)
resolveAsyncComponent @ vue.js:2326
createComponent @ vue.js:4145
_createElement @ vue.js:4357
createElement @ vue.js:4294
vm._c @ vue.js:4410
(anonymous) @ VM110:3
Vue._render @ vue.js:4465
updateComponent @ vue.js:2765
get @ vue.js:3113
Watcher @ vue.js:3102
mountComponent @ vue.js:2772
Vue$3.$mount @ vue.js:8416
Vue$3.$mount @ vue.js:10777
(anonymous) @ fullpage-component-loader.js:90
F @ _export.js:35
mountComponent @ fullpage-component-loader.js:72
(anonymous) @ fullpage-component-loader.js:43
Promise.then (async)
load @ fullpage-component-loader.js:42
(anonymous) @ index.js:95
Promise.then (async)
load @ index.js:94
load @ index.js:127
(anonymous) @ index.html:68
recorder-handlers.js:62 recorder unquiet event triggered
recorder-handlers.js:58 recorder quiet event triggered
recorder-handlers.js:62 recorder unquiet event triggered
recorder-handlers.js:58 recorder quiet event triggered 

Does this project support SAML authentication with ability to redirect to ADFS and back to the LEX bot.
or do we have to write that code or use something like cognito user pool.
